src/hotspot/share/gc/g1/g1Policy.cpp

```diff
@@ -1400,16 +1400,8 @@ uint G1Policy::calc_max_old_cset_length() const {
   // as a percentage of the heap size. I.e., it should bound the
   // number of old regions added to the CSet irrespective of how many
   // of them are available.
-
-  const G1CollectedHeap* g1h = G1CollectedHeap::heap();
-  const size_t region_num = g1h->num_regions();
-  const size_t perc = (size_t) G1OldCSetRegionThresholdPercent;
-  size_t result = region_num * perc / 100;
-  // emulate ceiling
-  if (100 * result < region_num * perc) {
-    result += 1;
-  }
-  return (uint) result;
+  // Round up to be conservative.
+  return (uint)ceil((double)_g1h->num_regions() * (size_t)G1OldCSetRegionThresholdPercent / 100);
```

Comment:
The type cast made this line unnecessarily harder to parse. Maybe break into multiple ones? Sth like:

```c
double ratio = ...
double result = ceil(...)
return (uint) result;
```